src/schema/v2/schema.ts

```diff
@@ -254,6 +255,7 @@ const rootFields = {
   artwork: Artwork,
   artworkAttributionClasses: ArtworkAttributionClasses,
   artworkMediums: ArtworkMediums,
+  artworkOrError: ArtworkOrError,
```

Comment:
Ideally we don't have to add new fields, and our root types would be able to accomodate. For example:

```
artwork(id: "foo") {
  slug
}

artwork(id: "foo") {
  ... on Artwork {
    slug
  }
  
  ... on PageError {
    statusCode 
  }
}
```

Reply:
I would tend to agree, but that's a breaking change for clients. Switching to a union type like this requires a new field, so that's why I went with the `orError`.

Reply:
This would be a breaking change? The union bits are additive:  

```
artwork(id: "foo") {
  slug 

  ... on Artwork {
    slug
  }
  
  ... on ArtworkError {
    slug 
  }
  
  ... on PageError {
    statusCode 
  }
}
```

Reply:
I kind of get what you're saying, but mechanically it is a breaking change unfortunately. If you switched `artwork` (like in your example) to be a union type, you can _only_ query it via fragments (`... on Artwork`) -> so all clients would break as they don't use that.

The only acceptable way to query it would be-

``` graphql
artwork(id: "foo") {
  ... on Artwork {
    slug
  }
  ...
}
```

so that's why we do need a new field, we can't really mix-in or decorate the existing.

Reply:
Can we really not do that? I can swear i've used fields like this in the past. But maybe i could be misremembering.

Reply:
Can we really not do that? I can swear i've used fields like this in the past. But maybe i could be misremembering.

Reply:
That's always been the case AFAIK. I view the `orError` pattern (while a 'touch' more verbose than regular `artwork`), as akin to preferring connections over lists. It's a touch more complex an object, but is more forward-looking and flexible in supporting other decorations and metadata that will come in the future.

That's why I'm sort of thinking we maybe could have had `artworkOrError` _instead of_ `artwork` all along as an optimal schema!

Reply:
How about `error` being a field on `artwork`?

Reply:
(also, `artwork` will error and return `"artwork": null` in the response, so you wouldn't be able to query for any top-level fields on `artwork`, like `error`)

Reply:
@damassi it would have been awesome if that was possible. I also think it's a breaking change, unfortunately. 
I would as well vouch for a union type here instead of an error field.

Reply:
I can't think of it in a way that won't break clients too. Union type is betters, is something we already do in the orders context in Exchange.

Reply:
Going to resolve this convo since I _think_ we're in agreement - union types are great(!), and going with a new root field for this. Will be a fun trial!